app/src/main/java/fr/free/nrw/commons/mwapi/ApacheHttpClientMediaWikiApi.java

```diff
@@ -982,46 +984,16 @@ public boolean isUserBlockedFromCommons() {
             Request request = new Request.Builder()
                     .url(urlBuilder.toString())
                     .build();
-            OkHttpClient client = new OkHttpClient();
-            Response response = client.newCall(request).execute();
-            String jsonData = response.body().string();
-            JSONObject jsonObject = new JSONObject(jsonData);
-            return jsonObject;
+            Response response = okHttpClient.newCall(request).execute();
```

Comment:
The issue seems to be caused by a SocketTimeoutException here. Now that the method is Nullable, should we catch that exception and return null as well?

Reply:
Thanks for pointing it out @misaochan. The consumers are handling `throwable` so explicit handling of `SocketTimeoutException` isn't required. Earlier the consumers were expecting a `@Nonnull` value but now the expectation has changed.